src/server/controllers/checkPasswordToken.ts

```diff
@@ -118,15 +152,34 @@ export const checkTokenInOkta = async (
     const timeUntilTokenExpiry =
       expiresAt && Date.parse(expiresAt) - Date.now();
 
-    setEncryptedStateCookie(res, { email, stateToken });
+    updateEncryptedStateCookie(req, res, { email, stateToken });
 
     trackMetric('OktaValidatePasswordToken::Success');
 
+    // since we can't pass query parameters through okta emails, we set the encryptedStateCookie
+    // as the email was sent to the user containing the query params at that time
+    // so we read them here, if the user comes back on the same browser
+    const encryptedState = readEncryptedStateCookie(req);
```

Comment:
Are these functions only ever going to be used for this specific cookie `GU_GATEWAY_STATE`? Are we using the cookie more generally, or only for this specific flow, during registration?

Reply:
The encrypted state cookie (`GU_GATEWAY_STATE`) is already being used is multiple flows, not just for this use case. We thought we could extend the functionality for this use case.

This particular use case it for okta registration and okta password reset methods.